Exclude system table systranschemas for mssql diff

MatrixDai opened the pull request:
<!--- This environment context section helps us quickly review your PR. 
      Please take a minute to fill-out this information. -->
## Environment

**Liquibase Version**: All

**Liquibase Integration & Version**: CLI

**Liquibase Extension(s) & Version**: NA

**Database Vendor & Version**: MSSQL

**Operating System Type & Version**: All

## Pull Request Type

<!--- What types of changes does your code introduce?
      Put an `x` in all the boxes that apply: 
      If this PR fixes an existing GH issue, edit the next line to add "closes #XXXX" to auto-link.
      If this PR fixes an existing CORE Jira issue, note that as well, although there will be no auto-linking. -->
- [ ] Bug fix (non-breaking change which fixes an issue.)
- [x] Enhancement/New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

Per this Microsoft document, the table `systranschemas` should belongs to system table/view.
https://docs.microsoft.com/en-us/sql/relational-databases/system-views/systranschemas-transact-sql?view=sql-server-ver15


## Steps To Reproduce

Try to diff 2 mssql schema, one of them has `systranschemas`.

## Actual Behavior
The system table `systranschemas` will be reported as "Missing Table(s)" or "Unexpected Table(s)".

## Expected/Desired Behavior
The system table `systranschemas` should NOT be reported as "Missing Table(s)" or "Unexpected Table(s)".

## Screenshots (if appropriate)
<img width="260" alt="Screen Shot 2021-06-27 at 9 46 26 PM" src="https://user-images.githubusercontent.com/1191662/123546905-2bccbd80-d791-11eb-9937-ca16cfdc04da.png">


## Additional Context
The change only impact for MSSQL database engine.

## Fast Track PR Acceptance Checklist:
<!--- Completing these speeds up the acceptance of your pull request -->
<!--- Put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, just ask us in a comment. We're here to help! -->
- [x] Build is successful and all new and existing tests pass
- [ ] Added [Unit Test(s)](https://liquibase.jira.com/wiki/spaces/LB/pages/1274937609/How+to+Write+Liquibase+Core+Unit+Tests)
- [ ] Added [Integration Test(s)](https://liquibase.jira.com/wiki/spaces/LB/pages/1276608569/How+to+Write+Liquibase+Core+Integration+Tests)
- [ ] Added [Test Harness Test(s)](https://github.com/liquibase/liquibase-test-harness/pulls)
- [ ] Documentation Updated

## Need Help?
Come chat with us on our [discord channel](https://discord.com/channels/700506481111597066/700506481572839505)

molivasdat commented:
Hi @MatrixDai Thanks for creating this PR.
A member of the Liquibase team will take a look at your contribution and may suggest: 

- Changes based on [guidelines for PR submission](https://liquibase.jira.com/wiki/spaces/LB/pages/1274904896/How+to+Contribute+Code+to+Liquibase+Core)
- Additional tests ([unit ](https://liquibase.jira.com/wiki/spaces/LB/pages/1274937609/How+to+Write+Liquibase+Core+Unit+Tests)and/or [integration](https://liquibase.jira.com/wiki/spaces/LB/pages/1276608569/How+to+Write+Liquibase+Core+Integration+Tests)) and/or [Test Harness Tests](https://github.com/liquibase/liquibase-test-harness/pulls)
- Provide other feedback, like requesting changes. 

The PR will be prioritized according to our internal development and testing capacity.

We’ll let you know when it’s ready to move to the next step or if any changes are needed.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/liquibase/liquibase?pullRequest=1944) <br/>All committers have signed the CLA.

filipelautert committed:
Add MSSQL regression test for systranschemas

filipelautert reviewed (approved):
I rebased the branch onto current `master` and added regression coverage for the original fix.
I've no idea why this was ignored in the past.

MatrixDai committed:
Introduce new diff option '--diffColumnDefaultValueConstraintName' to allow exclude mssql generated default constraint name.

- Add DIFF_COLUMN_DEFAULT_VALUE_CONSTRAINT_NAME field to GlobalConfiguration (liquibase-standard)
- Add exclusion logic in ColumnComparator.findDifferences() after DIFF_COLUMN_ORDER check

rberezen commented:
@MatrixDai Tests fail — testIsSystemObject for systranschemas expects true but gets false.
cc @filipelautert

MatrixDai committed:
Add systranschemas to MSSQL systemTablesAndViews

MatrixDai commented:
> @MatrixDai Tests fail — testIsSystemObject for systranschemas expects true but gets false. cc @filipelautert

The test MSSQLDatabaseTest should pass now. Thanks.

rberezen commented:
The systranschemas test passes now, thanks @MatrixDai.

CI is failing on `LiquibaseCommandLineTest.help output` — the new `DIFF_COLUMN_DEFAULT_VALUE_CONSTRAINT_NAME` global config option changes the CLI help output, but the expected help text in the test hasn't been updated to include it. You'll need to update the expected output in `liquibase-cli/src/test/groovy/liquibase/integration/commandline/LiquibaseCommandLineTest.groovy` to fix this.

MatrixDai committed:
test: update expected help output for diffColumnDefaultValueConstraintName option

MatrixDai committed:
test: fix line wrapping of LIQUIBASE_DIFF_COLUMN_DEFAULT_VALUE_CONSTRAINT_NAME in expected help output

MatrixDai commented:
@rberezen 

The new test fix committed, local run the following UT passed:
```
./mvnw -pl liquibase-cli test \
  -Dtest="LiquibaseCommandLineTest#help output" \
  -Dsurefire.failIfNoSpecifiedTests=false
```
